src/focus.service.ts

```diff
@@ -509,6 +509,17 @@ export class FocusService {
   private isFocusable(el: HTMLElement) {
     const role = el.getAttribute('role');
     const tabIndex = el.tabIndex;
+    let visible = true;
+    for (let parent = el; parent !== this.root; parent = parent.parentElement) {
+      if (parent.classList.contains('hidden')) {
+        visible = false;
+        break;
+      }
+    }
+
+    if (!visible) {
+      return false;
+    }
```

Comment:
This is just a bug waiting to happen. getComputedStyle should be cheap to call if it's called in a single stack smartly, only on candidates elements. I'm uncomfortable approving the solution presented in this PR.